src/EPR.Calculator.Frontend.UnitTests/LocalAuthorityDisposalCostsControllerTests.cs

```diff
@@ -1,19 +1,212 @@
+            };
+
+            IConfiguration configuration = new ConfigurationBuilder()
+                .AddInMemoryCollection(inMemorySettings)
+                .Build();
+
+            var clientFactory = new HttpClientFactoryStub();
+
+            // Act
+            var result = await LocalAuthorityDisposalCostsController.GetHttpRequest(configuration, clientFactory);
+
+            // Assert is handled by ExpectedException
+        }
+
+        [TestMethod]
+        public async Task Index_SuccessfulResponse_ReturnsViewWithGroupedData()
+        {
+            var mockHttpMessageHandler = new Mock<HttpMessageHandler>();
+            mockHttpMessageHandler
+                   .Protected()
+                .Setup<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.IsAny<HttpRequestMessage>(),
+                    ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(JsonConvert.SerializeObject(MockData.GetLocalAuthorityDisposalCosts()))
+                });
+
+            var httpClient = new HttpClient(mockHttpMessageHandler.Object);
+
+            // Mock IHttpClientFactory
+            var mockHttpClientFactory = new Mock<IHttpClientFactory>();
+            mockHttpClientFactory
+                .Setup(_ => _.CreateClient(It.IsAny<string>()))
+                .Returns(httpClient);
+
+            // Arrange
+            var controller = new LocalAuthorityDisposalCostsController(GetConfigurationValues(), mockHttpClientFactory.Object);
+
+            // Act
+
             var result = controller.Index() as ViewResult;
+
+            // Assert
             Assert.IsNotNull(result);
             Assert.AreEqual(ViewNames.LocalAuthorityDisposalCostsIndex, result.ViewName);
+            Assert.IsInstanceOfType(result.Model, typeof(List<IGrouping<string, LocalAuthorityViewModel>>));
+        }
+
+        [TestMethod]
+        public void Index_WhenExceptionThrown_RedirectsToErrorPage()
+        {
+            // Arrange
+            var controller = new LocalAuthorityDisposalCostsController(null, null);
+
+            // Act
+            var result = controller.Index() as RedirectToActionResult;
+
+            // Assert
+            Assert.AreEqual("Index", result.ActionName);
+            Assert.AreEqual("StandardError", result.ControllerName);
+        }
+
+        private static IConfiguration GetConfigurationValues()
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.Test.json", optional: false, reloadOnChange: true)
+                .AddEnvironmentVariables();
+
+            IConfiguration config = builder.Build();
+
+            return config;
+        }
+
+        private class HttpClientFactoryStub : IHttpClientFactory
```

Comment:
Any idea why you are using a stub and not a mock here? Ideally we should use a mock here. Something like this
`httpClientFactory.Setup(_ => _.CreateClient(string.Empty)) 
        .Returns(httpClient).Verifiable();`